devlib/target.py

```diff
@@ -1071,6 +1072,38 @@ async def write_value(self, path, value, verify=True, as_root=True):
             else:
                 raise
 
+    @contextmanager
+    def make_temp(self, is_directory=True, directory='', prefix='devlib-test'):
+        """
+        Creates temporary file/folder on target and deletes it once it's done.
+
+        :param is_directory: Specifies if temporary object is a directory, defaults to True.
+        :type is_directory: bool, optional
```

Comment:
I do see `, optional` on https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html. Is that wrong doc for Sphinx?
How do you verify docstring?

Reply:
That seems to be the right doc, but I think that either the doc is wrong or it's actually the doc on how to contribute to Sphinx itself, which may use something else than plain autodoc. That syntax is part of the [numpy style](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard), which requires the napoleon Sphinx extension. Apparently that napoleon extension pre-processes the docstring before autodoc sees it.

>  How do you verify docstring? 

You build the doc and check for warnings (`lisa-doc-build` for LISA). I tried by modifying a docstring in LISA. Since devlib's doc is not configured with autodoc, building the doc won't do anything as nothing will touch those docstrings.

Either way, if the function accepts a type T or `None` (which is the standard way of making a parameter "optional" in the common understanding of the term[1]), then it's actually more accurate to state `T or None`. 

[1] You can make a parameter actually optional without any default value with the variable keyword argument syntax, but that's definitely not something you'd do on a daily basis.

Reply:
> You build the doc and check for warnings
Another way would be to setup a test sphinx project, which is very easy too. Or port devlib to use autodoc. It's not particularly difficult, just an invasive PR (all existing function documentations need to be moved back as docstrings). If you were to do that, it would probably take ~1week (or maybe less) and I think it would be reasonable to as @marcbonnici  to freeze merging any other stuff so you don't end up with conflicts while working on it.

Reply:
I will not go for updating entire devlib repo now :)
Would you be OK if I just remove `, optional` part from docstring? Otherwise, it does not seem trivial to verify docstrings' Sphinx compatibility in devlib.

Reply:
Yes just go for replacing `, optional` by `or None`